habitat_sim/agent/agent.py

```diff
@@ -142,9 +142,18 @@ def reconfigure(
         if reconfigure_sensors:
             self._sensors.clear()
             for spec in self.agent_config.sensor_specifications:
-                self._sensors.add(
-                    hsim.CameraSensor(self.scene_node.create_child(), spec)
-                )
+                self.add_sensor(spec, modify_agent_config=False)
+
+    def add_sensor(
```

Comment:
This should be a private method since it doesn't make sense for anyone to call it besides the simulator.

Reply:
Sounds good.